From 22616a6f45b33962f231232993e3b3cb97ecda52 Mon Sep 17 00:00:00 2001
From: Bartosz Golaszewski <brgl@bgdev.pl>
Date: Wed, 13 Jul 2022 15:45:22 +0200
Subject: [PATCH 1/2] setup.py: don't hardcode paths

The setup script was written for jetpack. This modifies it to work with
local sources.

Signed-off-by: Bartosz Golaszewski <brgl@bgdev.pl>
---
Upstream-Status: Inappropriate [yocto-specific]

 setup.py | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/setup.py b/setup.py
index 63dac42..bf43e48 100644
--- a/setup.py
+++ b/setup.py
@@ -18,23 +18,20 @@ __version__ = "0.0.1"
 #   Sort input source files if you glob sources to ensure bit-for-bit
 #   reproducible builds (https://github.com/pybind/python_example/pull/53)
 
-include_dirs = [ "/usr/src/jetson_multimedia_api/include"
-    , "/usr/src/jetson_multimedia_api/include/libjpeg-8b" ]
+include_dirs = [ "jetson_multimedia_api/usr/src/jetson_multimedia_api/include"
+    , "jetson_multimedia_api/usr/src/jetson_multimedia_api/include/libjpeg-8b" ]
         
 
-library_dirs = [ "/usr/local/cuda/lib64"
-        , "/usr/lib/aarch64-linux-gnu/tegra"]
-
 libraries = [ "nvargus_socketclient"
     , "nvjpeg"
     , "EGL"
     , "nvbuf_utils" ]
 
 
-nvidia_source = [ "/usr/src/jetson_multimedia_api/samples/common/classes/NvJpegEncoder.cpp",
-        "/usr/src/jetson_multimedia_api/samples/common/classes/NvElement.cpp",
-        "/usr/src/jetson_multimedia_api/samples/common/classes/NvElementProfiler.cpp",
-        "/usr/src/jetson_multimedia_api/samples/common/classes/NvLogging.cpp",
+nvidia_source = [ "jetson_multimedia_api/usr/src/jetson_multimedia_api/samples/common/classes/NvJpegEncoder.cpp",
+        "jetson_multimedia_api/usr/src/jetson_multimedia_api/samples/common/classes/NvElement.cpp",
+        "jetson_multimedia_api/usr/src/jetson_multimedia_api/samples/common/classes/NvElementProfiler.cpp",
+        "jetson_multimedia_api/usr/src/jetson_multimedia_api/samples/common/classes/NvLogging.cpp",
         ]
 
 sources = list(glob("src/*.cpp"))
@@ -48,7 +45,6 @@ ext_modules = [
         define_macros = [('VERSION_INFO', __version__)],
         include_dirs = include_dirs,
         libraries = libraries,
-        library_dirs = library_dirs
         ),
 ]
 
-- 
2.34.1

